core/src/main/java/io/kestra/core/events/CrudEvent.java

```diff
@@ -1,11 +1,20 @@
 package io.kestra.core.events;
 
+import io.micronaut.core.annotation.Nullable;
 import lombok.AllArgsConstructor;
 import lombok.Getter;
 
 @AllArgsConstructor
 @Getter
 public class CrudEvent<T> {
     T model;
+    @Nullable
+    T modelPrevious;
```

Comment:
```suggestion
    T previous;
```

Or previousModel if you want to be more explicit

Reply:
Moreover, using an Optional<T> would have been more elegant and avoid having 2 constructors and adding a bunch of if everywhere to switch from 2 param args constructor to 3 params arg.